src/tls_wrap.cc

```diff
@@ -375,6 +382,7 @@ Local<Value> TLSWrap::GetSSLError(int status, int* err, const char** msg) {
           *msg = buf;
         }
         static_cast<void>(BIO_reset(bio));
+        BIO_free_all(bio);
```

Comment:
After looking at deps/openssl/openssl/crypto/bio/bss_mem.c I'm guessing the answer is 'yes'.  Instead of blindly adding BIO_free_all calls everywhere, I suggest introducing a small RAII class that fixes the issue once and for all.